src/population.hpp

```diff
@@ -89,7 +92,7 @@ std::vector<T> _readSelection(const HighFive::DataSet& dset, const Selection& se
 
 template <typename T, typename std::enable_if<std::is_pod<T>::value>::type* = nullptr>
 std::vector<T> _readSelection(const HighFive::DataSet& dset, const Selection& selection) {
-    if (selection.ranges().empty()) {
+    if (selection.ranges().empty() || dset.getElementCount() == 0) {
```

Comment:
We'll need to keep track of the `dset.getElementCount() == 0` (which is a collective property) vs. `selection.ranges().empty()` (which is not) during the upcoming refactoring.

Reply:
ok